app/src/main/java/com/bleyl/recurrence/database/DatabaseHelper.java

```diff
@@ -119,6 +123,13 @@ public void onUpgrade(SQLiteDatabase database, int oldVersion, int newVersion) {
                     + COL_FRIDAY + " TEXT, "
                     + COL_SATURDAY + " TEXT) ");
         }
+
+        if (oldVersion < 4) {
+            database.execSQL("ALTER TABLE " + NOTIFICATION_TABLE + " ADD " + COL_NAG_TIMER + " INTEGER");
+            database.execSQL("ALTER TABLE " + NOTIFICATION_TABLE + " ADD " + COL_ACTIVE_STATE + " BOOLEAN");
+            database.execSQL("UPDATE " + NOTIFICATION_TABLE + " SET " + COL_NAG_TIMER + " = 0;");
+            database.execSQL("UPDATE " + NOTIFICATION_TABLE + " SET " + COL_ACTIVE_STATE + " = FALSE;");
```

Comment:
I didnt try it out, but I am quite sure that this is because of a wrong pointing to the value.
Currently assigning values from the database will use indexes to point to the column that holds the value. But as the index of a column can vary. E.g. when a new database will be created, then the column "COL_NAG_TIMER" is on index 4. But if a previous database is existing, the onOpgrade-sequence will be executed, where "COL_NAG_TIMER" will be appended to the existing table, and so its on index 10.

So you either take care about adding the new columns to the end (what will need a change in CREATE TABLE-command) and read out new values from index 10 and 11 or directly point to the correct column.

As an example:
Currently it looks like this
reminder.setId(cursor.getInt(0));
reminder.setTitle(cursor.getString(1));

But a better way would be sth like this:
reminder.setId(cursor.getInt(cursor.getColumnIndexOrThrow(COL_ID)));
reminder.setTitle(cursor.getString(cursor.getColumnIndexOrThrow(COL_TITLE)));



As I didnt try out the code yet, I am not 100% sure if this fixes the crash you are running into now, but at least it will take care about an altered table layout where the column index may be different.